linalg/constraints.hpp

```diff
@@ -295,22 +325,28 @@ class PenaltyConstrainedSolver : public ConstrainedSolver
    PenaltyConstrainedSolver(HypreParMatrix& A, HypreParMatrix& B,
                             double penalty_);
 
+   PenaltyConstrainedSolver(HypreParMatrix& A, HypreParMatrix& B,
+                            Vector& penalty_);
+
    ~PenaltyConstrainedSolver();
 
    void Mult(const Vector& x, Vector& y) const override;
 
    void SetOperator(const Operator& op) override
    { MFEM_ABORT("Operator cannot be reset!"); }
 
+   void SetPreconditioner(Solver& precond) override
+   { prec = &precond; }
+
 protected:
-   void Initialize(HypreParMatrix& A, HypreParMatrix& B);
+   void Initialize(HypreParMatrix& A, HypreParMatrix& B, HypreParMatrix& D);
```

Comment:
It would be great to have a doxygen /// coment here about what A, B, and D  represent.

Reply:
Sure, I can put in a comment here.

Reply:
Nice!